File: 14.longest-common-prefix.cpp
// @leet start
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
  string longestCommonPrefix(vector<string> &strs) {
    string result = "";
    bool matches = true;
    int n = 0;
    char prev,next;
    while (matches) {
		if (n<strs[0].size()) {
    	prev = strs[0][n];
		}else {
			break;
		}
      for (int i = 0; i < strs.size(); i++) { 
        next = strs[i][n];
				if (prev!=next) {
					matches=false;
					break;
				}
				else if (i==strs.size()-1 && matches==true) {
					result.push_back(prev);
				}
      }
			n++;
    }

    return result;
  }
};
// @leet end

File: 14.longest-common-prefix.js
// @leet start
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
	let answer = "";
	if (strs.length === 1) {
		return strs[0];
	}
	for (let i = 0; i < strs[0].length; i++) {
		for (let n = 0; n < strs.length; n++) {
			if (strs[0][i] !== strs[n][i]) {
				return answer;
			}
		}
		answer += strs[0][i];
	}
	return answer;
};
// @leet end
